pkg/controllers/restore/controller.go

```diff
@@ -369,6 +370,11 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 		}
 
 		metadata := resourceData.Object[metadataMapKey].(map[string]interface{})
+		// ignore the secrets section in the service account to avoid missing token secret error
+		if resourceData.Object["kind"].(string) == "ServiceAccount" {
```

Comment:
So this will actually get hit on every type of object, but then if the object type is a service account then we delete the secret that backs it ?

Reply:
correct.

Reply:
As discussed in chat, type assertion is unnecessary here. In fact it may result in panic is somehow it's not string or nil.